src/OpenTelemetry/Resources/Resources.cs

```diff
@@ -24,40 +24,50 @@ public static class Resources
     {
         /// <summary>
         /// Creates a new <see cref="Resource"/> from service information following standard convention
-        /// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md#service. 
+        /// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md#service.
         /// </summary>
         /// <param name="serviceName">Name of the service.</param>
         /// <param name="serviceInstanceId">Unique identifier of the service instance.</param>
         /// <param name="serviceNamespace">Optional namespace of the service.</param>
         /// <param name="serviceVersion">Optional version of the service.</param>
-        public static Resource CreateServiceResource(string serviceName, string serviceInstanceId = null, string serviceNamespace = null, string serviceVersion = null)
+        /// <param name="attributes">Optional additional service attributes.</param>
+        public static Resource CreateServiceResource(
+            string serviceName,
+            string serviceInstanceId = null,
+            string serviceNamespace = null,
+            string serviceVersion = null,
+            IDictionary<string, object> attributes = null)
```

Comment:
I understand the intent behind adding optional attributes. At the same time we have Resource.Merge for it and it also enforces some requirements on the attributes.

I suggest that we keep API as slim as possible at this point and remove optional attributes.

Reply:
I like this:

```csharp
                builder
                    .SetResource(Resources.CreateServiceResource(
                        "MyServiceName",
                        serviceVersion: "2.0.1",
                        attributes: new Dictionary<string, object>
                        {
                            ["datacenter.id"] = 1,
                        }));
```

Much more than this:

```csharp
                builder
                    .SetResource(
                        Resources
                            .CreateServiceResource(
                                "MyServiceName",
                                serviceVersion: "2.0.1")
                            .Merge(new Resource(new Dictionary<string, object>
                                {
                                    ["datacenter.id"] = 1,
                                })));
```

You still want me to remove?

Reply:
yes please remove, it's not because current state is better, it's because we need much better story around it, but it's not a first priority. Both options are kind of ugly.

(Better story is: 'Set up for K8S' auto-detecting everything that is possible)

Reply:
Done